pkg/endpoint/bpf.go

```diff
@@ -242,31 +251,152 @@ func (e *Endpoint) addNewRedirectsFromMap(m policy.L4PolicyMap, desiredRedirects
 	return nil, finalizeList.Finalize, revertStack.Revert
 }
 
+func (e *Endpoint) addVisibilityRedirects(ingress bool, desiredRedirects map[string]bool, proxyWaitGroup *completion.WaitGroup) (error, revert.FinalizeFunc, revert.RevertFunc) {
+	var (
+		visPolicy               policy.DirectionalVisibilityPolicy
+		direction               trafficdirection.TrafficDirection
+		policyEnabled           bool
+		finalizeList            revert.FinalizeList
+		revertStack             revert.RevertStack
+		updatedStats            []*models.ProxyStatistics
+		insertedDesiredMapState = make(map[policy.Key]struct{})
+	)
+
+	if e.visibilityPolicy == nil {
+		return nil, finalizeList.Finalize, revertStack.Revert
+	}
+
+	if ingress {
+		visPolicy = e.visibilityPolicy.Ingress
+		direction = trafficdirection.Ingress
+		policyEnabled = e.desiredPolicy.IngressPolicyEnabled
+	} else {
+		visPolicy = e.visibilityPolicy.Egress
+		direction = trafficdirection.Egress
+		policyEnabled = e.desiredPolicy.EgressPolicyEnabled
+	}
+
+	if !policyEnabled {
```

Comment:
If `policyEnable` is true we should not even continue at all, including doing a `revertStack.Push` in line 335